bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/InboxAutoIgnore.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2014-2017 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.config.discovery.internal;
+
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultFlag;
+import org.eclipse.smarthome.config.discovery.inbox.Inbox;
+import org.eclipse.smarthome.config.discovery.inbox.InboxFilterCriteria;
+import org.eclipse.smarthome.core.events.AbstractTypedEventSubscriber;
+import org.eclipse.smarthome.core.events.EventSubscriber;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingRegistry;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.events.ThingStatusInfoChangedEvent;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class implements a service to automatically ignore {@link Inbox} entries of newly created things.
+ * <p>
+ * The {@link InboxAutoIgnore} service implements a {@link EventSubscriber}, that is triggered
+ * for each thing when coming ONLINE. {@link Inbox} entries with the same device id like the
+ * newly created thing will be automatically set to {@link DiscoveryResultFlag#IGNORED}.
+ * </p>
+ * <p>
+ * If a thing is being removed, possibly existing {@link Inbox} entries with the same device id
+ * are removed from the {@link Inbox} so they could be discovered again afterwards.
+ * </p>
+ * <p>
+ * This service can be enabled or disabled by setting the {@code autoIgnore} property to either
+ * {@code true} or {@code false} via ConfigAdmin.
+ * </p>
+ *
+ * @author Andre Fuechsel - Initial Contribution
+ */
+@Component(immediate = true, service = EventSubscriber.class, property = {
+        "service.config.description.uri=system:inbox", "service.config.label=Inbox Auto Ignore",
```

Comment:
Service meta-data for the UI should not be on a single feature level - imho it would be better to simply have a "Inbox" entry (instead of "Inbox Auto Ignore"), which could offer multiple things that might be possible to do in the context of the inbox.
There is e.g. already an ["Auto Approve" service](https://github.com/openhab/openhab-core/blob/master/bundles/org.openhab.core/src/main/java/org/openhab/core/internal/inbox/AutoApproveService.java) in openHAB - it might make sense to move that (very small) feature into your service here.